compiler/damlc/lib/DA/Cli/Damlc/DependencyDb.hs

```diff
@@ -0,0 +1,204 @@
+depMarkerFile depPath = depPath </> "_dependency_"
+dataDepMarkerFile depPath = depPath </> "_data_dependency_"
+
+
+
+-- Dependency installation
+--------------------------
+-- | Install all dependencies to the .daml/dependencies directory.
+installDependencies ::
+       NormalizedFilePath -> Options -> PackageSdkVersion -> [String] -> [String] -> IO ()
+installDependencies projRoot opts sdkVer@(PackageSdkVersion thisSdkVer) pDeps pDataDeps = do
+    deps <- expandSdkPackages (optDamlLfVersion opts) (filter (`notElem` basePackages) pDeps)
+    (needsUpdate, newFingerprint) <-
+        depsNeedUpdate depsDir (deps ++ pDataDeps) thisSdkVer (show $ optDamlLfVersion opts)
+    when needsUpdate $ do
+        removePathForcibly depsDir
+        createDirectoryIfMissing True depsDir
+        -- install dependencies
+        -----------------------
+        depsExtracted <- mapM extractDar deps
+        checkSdkVersions sdkVer depsExtracted
+        forM_ depsExtracted $ installDar depsDir False
+        -- install data-dependencies
+        ----------------------------
+        let (fpDars, fpDalfs) = partition ((== ".dar") . takeExtension) pDataDeps
+        forM_ fpDars $ extractDar >=> installDar depsDir True
+        forM_ fpDalfs $ installDataDepDalf depsDir
+        -- write new fingerprint
+        write (fingerprintFile depsDir) $ encode newFingerprint
+  where
+    depsDir = dependenciesDir opts projRoot
+
+-- | Check that only one sdk version is present in dependencies and it equals this sdk version.
+checkSdkVersions :: PackageSdkVersion -> [ExtractedDar] -> IO ()
+checkSdkVersions (PackageSdkVersion thisSdkVer) depsExtracted = do
+    let uniqSdkVersions = nubSort $ thisSdkVer : map edSdkVersions depsExtracted
+    let depsSdkVersions = map edSdkVersions depsExtracted
+    unless (all (== thisSdkVer) depsSdkVersions) $
+        fail $
+        "Package dependencies from different SDK versions: " ++ intercalate ", " uniqSdkVersions
+
+-- Install a dar dependency
+installDar :: FilePath -> Bool -> ExtractedDar -> IO ()
+installDar depsPath isDataDep ExtractedDar {..} = do
+    let bs = BSL.toStrict $ ZipArchive.fromEntry edMain
+    let fp = ZipArchive.eRelativePath edMain
+    DecodedDalf {decodedUnitId, decodedDalfPkg} <- either fail pure $ decodeDalf Set.empty fp bs
+    let relDepPath =
+            unitIdString decodedUnitId <> "-" <>
+            (T.unpack $ LF.unPackageId $ LF.dalfPackageId decodedDalfPkg)
+    let depPath = depsPath </> relDepPath
+    if isDataDep
+        then write (dataDepMarkerFile depPath) ""
+        else write (depMarkerFile depPath) ""
+    let path = mainDir depPath </> (takeFileName $ ZipArchive.eRelativePath edMain)
+    write path (ZipArchive.fromEntry edMain)
+    forM_ edConfFiles $ \conf ->
+        write
+            (configDir depPath </> (takeFileName $ ZipArchive.eRelativePath conf))
+            (ZipArchive.fromEntry conf)
+    forM_ edSrcs $ \src ->
```

Comment:
What happens if we don’t extract those? In other words, can we just drop them?

Reply:
`Message: 
  daml/Main.daml:7:1: error:
  Could not find module ‘Daml.Script’
  There are files missing in the ‘daml-script-0.0.0’ package,
  try running 'ghc-pkg check'.
  Use -v (or `:set -v` in ghci) to see a list of the files searched for.
`
I think you need them.

Reply:
Oh does `edSrcs` include iface files and not just source files?

Reply:
yes, and .hie files.